### PR TITLE
Feature/programm options new vector type

### PR DESCRIPTION
**Pull request by dothebart**
### Scope & Purpose
This adds the ability to have vectors of predefined values in the config object, which is needed in later PRs.

- [x] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

#### Related Information
this is a subset of https://github.com/arangodb/arangodb/pull/10070

### Testing & Verification

This PR adds tests that were used to verify all changes:

- [x] Added new C++ **Unit Tests** (Either GoogleTest or Catch-Test)

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6410/

File: tests/Basics/InifileParserTest.cpp [hunk 1]
```diff
@@ -98,6 +102,14 @@ TEST(InifileParserTest, test_options) {
   options.addOption("--pork.a-string-value-with-an-inline-comment", "bla", new StringParameter(&aStringValueWithAnInlineComment));
   options.addOption("--pork.another-string-value-with-an-inline-comment", "bla", new StringParameter(&anotherStringValueWithAnInlineComment));
   options.addOption("--pork.a-string-value-not-set", "bla", new StringParameter(&aStringValueNotSet));
+  options.addOption("--pork.sounds", "which sounds do porks make?",
+                    new DiscreteValuesVectorParameter<StringParameter>(&porkSounds, soundsPorksMake),
+                    arangodb::options::makeFlags(options::Flags::FlushOnFirst));
+
+  options.addOption("--pork.strange-sounds", "which strange sounds do porks make?",
```
Comment (jsteemann) [inline tests/Basics/InifileParserTest.cpp]:
```suggestion
  options.addOption("--pork.strange-sounds", "which strange sounds do pigs make?",
```

File: tests/Basics/InifileParserTest.cpp [hunk 2]
```diff
@@ -98,6 +102,14 @@ TEST(InifileParserTest, test_options) {
   options.addOption("--pork.a-string-value-with-an-inline-comment", "bla", new StringParameter(&aStringValueWithAnInlineComment));
   options.addOption("--pork.another-string-value-with-an-inline-comment", "bla", new StringParameter(&anotherStringValueWithAnInlineComment));
   options.addOption("--pork.a-string-value-not-set", "bla", new StringParameter(&aStringValueNotSet));
+  options.addOption("--pork.sounds", "which sounds do porks make?",
```
Comment (jsteemann) [inline tests/Basics/InifileParserTest.cpp]:
```suggestion
  options.addOption("--pork.sounds", "which sounds do pigs make?",
```

File: lib/ProgramOptions/ProgramOptions.h [hunk 3]
```diff
@@ -267,6 +267,8 @@ class ProgramOptions {
   std::string _more;
   // context string that's shown when errors are printed
   std::string _context;
+  // already seen to flush programm options
+  std::vector<std::string> _seenParams;
```
Comment (jsteemann) [inline lib/ProgramOptions/ProgramOptions.h]: Should we rename this to `_alreadyFlushed` and change it to a set/unordered_set?

File: lib/ProgramOptions/ProgramOptions.h [hunk 4]
```diff
@@ -267,6 +267,8 @@ class ProgramOptions {
   std::string _more;
   // context string that's shown when errors are printed
   std::string _context;
+  // already seen to flush programm options
```
Comment (jsteemann) [inline lib/ProgramOptions/ProgramOptions.h]:
```suggestion
  // already seen to flush program options
```

**Comment by dothebart** [comment]
tests green.

**Pull request merged**